packages/compass-collection/src/plugin-tab-title.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import toNS from 'mongodb-ns';
+import {
+  useConnectionInfo,
+  useConnectionsListRef,
+  useTabConnectionTheme,
+} from '@mongodb-js/compass-connections/provider';
+import {
+  WorkspaceTab,
+  type WorkspaceTabCoreProps,
+} from '@mongodb-js/compass-components';
+import type { WorkspacePluginProps } from '@mongodb-js/compass-workspaces';
+
+import { type CollectionState } from './modules/collection-tab';
+
+export const CollectionWorkspaceTitle = 'Collection' as const;
+
+type PluginTitleProps = {
+  isTimeSeries?: boolean;
+  isReadonly?: boolean;
+  sourceName?: string | null;
+} & WorkspaceTabCoreProps &
+  WorkspacePluginProps<typeof CollectionWorkspaceTitle>;
+
+function _PluginTitle({
+  editViewName,
+  isNonExistent,
+  isReadonly,
+  isTimeSeries,
+  sourceName,
+  namespace,
+  ...tabProps
+}: PluginTitleProps) {
+  const { getThemeOf } = useTabConnectionTheme();
+  const { getConnectionById } = useConnectionsListRef();
+  const { id: connectionId } = useConnectionInfo();
+
+  const { database, collection, ns } = toNS(namespace);
+  const connectionName = getConnectionById(connectionId)?.title || '';
```

Comment:
There's a small issue with `useConnectionInfo` implementation (and so the types are correctly hiding that `title` is already returned here due to the test value fallback), but you can clean this up a but and just access title from connectionInfo returned by the hook (but feel free to keep this out of this PR to avoid more changes piling up)

Reply:
Good callout. Tests are green now, I'll do that in a follow up!